pdns/dnsdist-lua-actions.cc

```diff
@@ -273,19 +273,28 @@ void TeeAction::worker()
 class PoolAction : public DNSAction
 {
 public:
-  PoolAction(const std::string& pool) : d_pool(pool) {}
+  PoolAction(const std::string& pool, bool stopProcessing) : d_pool(pool), d_stopProcessing(stopProcessing) {}
+
   DNSAction::Action operator()(DNSQuestion* dq, std::string* ruleresult) const override
   {
-    *ruleresult=d_pool;
-    return Action::Pool;
+    if (d_stopProcessing) {
```

Comment:
Depending on the flag `dq.poolname` is set via `ruleresult` or directly. I find that a bit inconsistent.
Maybe always set it in the action and leave out the assignment in `applyRulesToQuery`? (Don't forget to fix `QPSPoolAction` as wel if you go that route.

Reply:
That would be much better but I can't leave the assignment out because a custom Lua action can return ``DNSAction.Spoof`, <value to spoof>`. I guess we will have to deprecate that syntax someday because it's not the first time it leads to pain.

Reply:
I guess you mean `Action::Pool`, but OK, a comment is in place then. Another question: shouldn't `QPSPoolAction` get the same optional arg?

Reply:
Ugh, yes, `DNSAction.Pool`! You are right, I only looked at `PoolAction` because that was what I needed at the moment but I'll see if more actions could use the same option.